src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -16717,5 +16717,862 @@ partial class Point(int i)
+            comp.VerifyDiagnostics(
+                // (2,22): warning CS9113: Parameter 'p1' is unread.
+                // partial class C1(int p1) : Base;
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p1").WithArguments("p1").WithLocation(2, 22),
+                // (8,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(8, 20)
+                );
+
+            comp = CreateCompilation(source2 + source1 + source3, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics(
+                // (6,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(6, 20),
+                // (10,22): warning CS9113: Parameter 'p1' is unread.
+                // partial class C1(int p1) : Base;
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p1").WithArguments("p1").WithLocation(10, 22)
+                );
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_25_OrderOfBinding()
+        {
+            var source1 = @"
+partial class C1(int p1) : Base("""");
+";
+            var source2 = @"
+partial class C1
+{
+    void M()
+    {
+        string a = p1; 
+    }
+}
+";
+            var source3 = @"
+class Base(string p1)
+{
+    protected string p1 = p1;
+}
+";
+            var comp = CreateCompilation(new[] { source1, source2, source3 }, options: TestOptions.ReleaseDll);
+            var tree = comp.SyntaxTrees[1];
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+
+            model.GetDiagnostics().Verify(
+                // 1.cs(6,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(6, 20)
+                );
+
+            comp.VerifyDiagnostics(
+                // 0.cs(2,22): warning CS9113: Parameter 'p1' is unread.
+                // partial class C1(int p1) : Base;
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p1").WithArguments("p1").WithLocation(2, 22),
+                // 1.cs(6,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(6, 20)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are there tests for the following?
- shadowing by a member of the base of the base
- shadowing with a property

Reply:
> Are there tests for the following?

Added ShadowedByMemberFromBase_26_PropertyVsParameterInIndirectBase.